apps/test/unit/code-studio/components/libraries/LibraryPublisherTest.js

```diff
@@ -0,0 +1,286 @@
+import {expect} from '../../../../util/reconfiguredChai';
+import React from 'react';
+import {shallow} from 'enzyme';
+import LibraryPublisher, {
+  PublishState
+} from '@cdo/apps/code-studio/components/libraries/LibraryPublisher';
+import LibraryClientApi from '@cdo/apps/code-studio/components/libraries/LibraryClientApi';
+import libraryParser from '@cdo/apps/code-studio/components/libraries/libraryParser';
+import sinon from 'sinon';
+import {replaceOnWindow, restoreOnWindow} from '../../../../util/testUtils';
+
+describe('LibraryPublisher', () => {
+  let publishSpy, clientApi, libraryDetails;
+  let defaultSelectedFunctions, sourceFunctionList;
+  const defaultDescription = '';
+  const librarySource = '//comment\nfunction foo(){}';
+  const libraryName = 'libraryName';
+  const CHECKBOX_SELECTOR = 'input[type="checkbox"]';
+
+  before(() => {
+    replaceOnWindow('dashboard', {
+      project: {
+        setLibraryName: () => {},
+        setLibraryDescription: () => {}
+      }
+    });
+    sinon.stub(window.dashboard.project, 'setLibraryName').returns(undefined);
+    sinon
+      .stub(window.dashboard.project, 'setLibraryDescription')
+      .returns(undefined);
+    sinon.stub(libraryParser, 'suggestName').returns(libraryName);
+    sinon.stub(libraryParser, 'sanitizeName').returns(libraryName);
+    clientApi = new LibraryClientApi('123');
+    publishSpy = sinon.stub(clientApi, 'publish');
+  });
+
+  after(() => {
+    libraryParser.suggestName.restore();
+    libraryParser.sanitizeName.restore();
+    restoreOnWindow('dashboard');
+  });
+
+  let onPublishSuccess, onUnpublishSuccess;
+  beforeEach(() => {
+    defaultSelectedFunctions = {};
+    sourceFunctionList = [{functionName: 'foo', comment: 'comment'}];
+    libraryDetails = {
+      libraryName: libraryName,
+      libraryDescription: defaultDescription,
+      selectedFunctions: defaultSelectedFunctions,
+      sourceFunctionList: sourceFunctionList,
+      librarySource: librarySource
+    };
+    onPublishSuccess = sinon.stub();
+    onUnpublishSuccess = sinon.stub();
+  });
+
+  afterEach(() => {
+    onPublishSuccess.resetHistory();
+    onUnpublishSuccess.resetHistory();
+  });
+
+  describe('On Load', () => {
+    it('suggests a default library name', () => {
+      libraryDetails.libraryName = 'foo';
+      let wrapper = shallow(
+        <LibraryPublisher
+          onPublishSuccess={onPublishSuccess}
+          onUnpublishSuccess={onUnpublishSuccess}
+          libraryDetails={libraryDetails}
+          clientApi={clientApi}
+        />
+      );
+
+      expect(wrapper.state().libraryName).to.equal(libraryName);
```

Comment:
since this piece of state is rendered by the component, i think it would be more useful here to test what is rendered  rather than the internal state of the component (similar to what you're doing on lines 93-94). something like:

```javascript
const libNameInput = wrapper.find('input').first();  // this is probably wrong, but you get it
expect(libNameInput.text()).to.equal(libraryName);
```

Reply:
✔️